rust/processor/src/db/common/models/events_models/parquet_events.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(clippy::extra_unused_lifetimes)]
+
+use crate::{
+    bq_analytics::generic_parquet_processor::{GetTimeStamp, HasVersion, NamedTable},
+    utils::util::{standardize_address, truncate_str},
+};
+use allocative_derive::Allocative;
+use aptos_protos::transaction::v1::{Event as EventPB, EventSizeInfo};
+use itertools::Itertools;
+use parquet_derive::ParquetRecordWriter;
+use serde::{Deserialize, Serialize};
+
+// p99 currently is 303 so using 300 as a safe max length
+const EVENT_TYPE_MAX_LENGTH: usize = 300;
+
+#[derive(Allocative, Clone, Debug, Default, Deserialize, ParquetRecordWriter, Serialize)]
+pub struct Event {
+    pub sequence_number: i64,
+    pub creation_number: i64,
+    pub account_address: String,
+    pub txn_version: i64,
+    pub block_height: i64,
+    pub event_type: String,
+    pub data: String,
+    pub event_index: i64,
+    pub indexed_type: String,
+    pub type_tag_bytes: i64,
+    pub total_bytes: i64,
```

Comment:
@ying-w could you check if I am not missing any fields?

Reply:
for future proofing, would be nice to have a event_version: i8 and set it to 1 and change it when events v2 comes out

what's the difference between type and indexed_type?

Reply:
can we re-order so that txn_version, account_address is first